subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -1380,6 +1381,15 @@ private void validate(String output, String displayName) {
                 }
             }
 
+            private boolean removeFirstExpectedDeprecationWarning(final Predicate<String> condition) {
```

Comment:
I don't think there was/is a reason for the remove all behavior - I think it just wasn't a problem since there was only 1 deprecation of any given type tested for at once.  This change supports duplicates, where the same deprecation warning text appears multiple times within a single build.  This is needed now with the new Options configuration warning added in this PR.